test/e2e/validation_test.go

```diff
@@ -464,6 +465,68 @@ func testExpectedServicesRunning(t *testing.T) {
 	}
 }
 
+// testServicesConfigMap tests 2 aspects of expected functionality for the services ConfigMap
+// 1. It exists on operator startup and 2. It is recreated when deleted
+func testServicesConfigMap(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	operatorVersion, err := getWMCOVersion()
+	require.NoError(t, err)
+	// There is no easy way to populate ldflags when running e2e tests so that version.Get() will return
+	// anything other than "" without having to maintain the WMCO version in another location
+	servicesConfigMapName := servicescm.ServicesConfigMapName + operatorVersion
+
+	err = tc.validateWindowsServicesConfigMap(servicesConfigMapName)
+	require.NoError(t, err, "error validating windows-services ConfigMap")
+
+	err = tc.deleteAndWaitForServicesConfigMap(servicesConfigMapName)
+	require.NoError(t, err, "error ensuring windows-services ConfigMap is re-created when deleted")
+}
+
+// validateWindowsServicesConfigMap ensures the windows-services ConfigMap exists in the cluster and has valid contents
+func (tc *testContext) validateWindowsServicesConfigMap(cmName string) error {
+	windowsServices, err := tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Get(context.TODO(),
+		cmName, meta.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "error retrieving ConfigMap: %s", cmName)
+	}
+
+	if _, err := servicescm.Parse(windowsServices.Data); err != nil {
+		return errors.Wrap(err, "invalid windows-services ConfigMap data")
+	}
+	return nil
+}
+
+// deleteAndWaitForServicesConfigMap deletes the given ConfigMap and ensures it is re-created properly
+func (tc *testContext) deleteAndWaitForServicesConfigMap(cmName string) error {
+	err := tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Delete(context.TODO(), cmName,
+		meta.DeleteOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "error deleting ConfigMap %s/%s", tc.namespace, cmName)
+	}
+
+	cm := &core.ConfigMap{}
+	err = wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
+		cm, err = tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Get(context.TODO(), cmName, meta.GetOptions{})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return false, errors.Wrapf(err, "error retrieving ConfigMap: %s", cmName)
+			}
+		}
+		// retry until timeout if the error is a resource NotFound error
+		return err == nil, nil
+	})
+	if err != nil {
+		return errors.Wrapf(err, "error waiting for ConfigMap %s/%s to be re-created", tc.namespace, cmName)
+	}
+
+	if _, err := servicescm.Parse(cm.Data); err != nil {
```

Comment:
@aravindhp you'd [previously commented here](https://github.com/openshift/windows-machine-config-operator/pull/1002#discussion_r831502015)

> What is the advantage in calling this function given this will be called by the operator code also? The real check should be if the operator is not throwing an error.

Are you suggesting to remove this `Parse()` call? The goal here is to ensure it is recreated in valid manner

Reply:
Correct. Rather than using `Parse()` you should check for the signal or lack of that an incorrect ConfigMap is present.